packages/devtools_app/lib/main.dart

```diff
@@ -4,21 +4,31 @@
 
 import 'package:flutter/material.dart';
 import 'package:flutter/foundation.dart';
-import 'package:html_shim/html.dart' show document;
+import 'package:html_shim/html.dart' show document, window;
 
 import 'src/flutter/app.dart';
+import 'src/framework/framework_core.dart';
 
 void main() {
+  String url;
   if (kIsWeb) {
     // Clear out the unneeded HTML from index.html.
     for (var element in document.body.querySelectorAll('.legacy-dart')) {
       element.remove();
     }
+    url = window.location.toString();
   } else {
     // When running in a desktop embedder, Flutter throws an error because the
     // platform is not officially supported. This is not needed for web.
     debugDefaultTargetPlatformOverride = TargetPlatform.fuchsia;
+    // TODO(jacobr): we don't yet have a direct analog to the URL on flutter
```

Comment:
Flutter's routing system is the cross-platform baseline that we have available to us.

To work well cross-platform, I believe we'll need to make routes and web URLs roughly isomorphic.

Reply:
I agree that is a good long term goal